core/domain/exp_domain_test.py

```diff
@@ -344,6 +344,40 @@ def test_validation(self):
                 utils.ValidationError, 'destination ABC is not a valid'):
             exploration.validate()
 
+        # Ensure an answer group with two fuzzy rules is invalid
+        init_state = exploration.states[exploration.init_state_name]
+        default_outcome = init_state.interaction.default_outcome
+        default_outcome.dest = exploration.init_state_name
```

Comment:
The bit restoring the exploration to a valid state should probably go above this section of code (similar to other convention in this file). It also helps separate the logic specific to this test from the housekeeping code for the exploration.